Adds hardcoded version and adds github workflow files

saltiyazan opened the pull request:
- Adds the `version` keyword to the snapcraft.yaml file and uses a hardcoded version.
- Updates the repository to use the shared Github Actions. (Snapcrafters/ci)

saltiyazan committed:
Adds hardcoded version and adds github workflow files
Uses shared actions from Snapcrafters

saltiyazan commented:
Hi @merlijn-sebrechts 
Here is my first PR to Snapcrafters, I followed the contributing guide to add the workflows.
Could you please take a look?

jnsgruk commented on .github/workflows/release-to-candidate.yml:
```diff
@@ -0,0 +1,72 @@
+        with:
+          architectures: ${{ needs.get-architectures.outputs.architectures }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  screenshots:
+    name: 📸 Gather screenshots
+    needs: call-for-testing
+    environment: "Candidate Branch"
+    runs-on: ubuntu-latest
+    steps:
+      - name: 📸 Gather screenshots
+        uses: snapcrafters/ci/get-screenshots@main
+        with:
+          issue-number: ${{ needs.call-for-testing.outputs.issue-number }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          screenshots-token: ${{ secrets.SNAPCRAFTERS_BOT_COMMIT }}
```
I don't think this is likely to be useful in a CLI application - I think it can be removed :)

saltiyazan replied:
✅

jnsgruk reviewed (dismissed):
LGTM - probably worth removing the `get-screenshot` action before we merge. Once you've got that sorted.

I'll add the credentials to the repo now.

jnsgruk commented:
Could you please also add a README using the template shown in https://github.com/snapcrafters/discord?

jnsgruk commented on snap/snapcraft.yaml:
```diff
@@ -3,11 +3,15 @@ adopt-info: get-iplayer
 confinement: strict
 assumes:
   - command-chain
+architectures:
+  - amd64
+  - arm64
```
```suggestion
architectures:
  - amd64
  - arm64
  - armhf
  - ppc64el
  - s390x
```

The snap currently has releases across all of these architectures :) I don't think anyone is actually using the `ppc64el` or `s390x` releases (according to the metrics), but we should probably leave `armhf` until it causes us problems.

saltiyazan replied:
✅

saltiyazan committed:
Removes screenshot taking step

saltiyazan committed:
Adds all the architectures on which the snap currently has releases

saltiyazan committed:
Adds a LICENSE file

saltiyazan committed:
Adds README

saltiyazan commented:
> Could you please also add a README using the template shown in https://github.com/snapcrafters/discord?

I added it, I'm not fully sure about the maintainers and license sections (I added the LICENSE file too) :)

saltiyazan committed:
Fixes typos

jnsgruk commented on README.md:
```diff
@@ -0,0 +1,87 @@
+<p align="center">
+<a href="https://snapcraft.io/get-iplayer"><img src="https://snapcraft.io/get-iplayer/badge.svg" alt="Snap Status"></a>
+<a href="https://github.com/snapcrafters/get-iplayer/actions/workflows/sync-version-with-upstream.yml"><img src="https://github.com/snapcrafters/get-iplayer/actions/workflows/sync-version-with-upstream.yml/badge.svg"></a>
+<a href="https://github.com/snapcrafters/get-iplayer/actions/workflows/release-to-candidate.yaml"><img src="https://github.com/snapcrafters/get-iplayer/actions/workflows/release-to-candidate.yaml/badge.svg"></a>
```
```suggestion
<a href="https://github.com/snapcrafters/get-iplayer/actions/workflows/release-to-candidate.yml"><img src="https://github.com/snapcrafters/get-iplayer/actions/workflows/release-to-candidate.yml/badge.svg"></a>
```

jnsgruk committed:
Update README.md

jnsgruk commented:
Thanks @saltiyazan!
